algolia.js

```diff
@@ -13,36 +13,44 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 require('dotenv').config();
+
 const algoliasearch = require('algoliasearch');
-const byteof = require('byteof');
 const fs = require('fs');
 const path = require('path');
+const truncateUTF8Bytes = require('truncate-utf8-bytes');
 
+const {sizeOfJSONInBytes} = require('./shared/sizeOfJSONInBytes');
 const {defaultLocale, supportedLocales} = require('./shared/locale');
 
 const maxChunkSizeInBytes = 10000000; // 10,000,000
 const maxChunkSizeInMB = maxChunkSizeInBytes / 1000000;
-const maxItemSizeInBytes = 7000; // 7,000
+// This limit is set by our current Algolia plan (with some headroom).
+// The restriction applies to the size of the JSON serialization of each item.
+// See https://support.algolia.com/hc/en-us/articles/4406981897617-Is-there-a-size-limit-for-my-index-records-#:~:text=The%20record%20size%20limit%20is%20based%20on%20the%20size%20of%20this%20final%20JSON%20file.
+const maxItemSizeInBytes = 9000;
```

Comment:
This was previously set to 7000, even though the API has a max of ~10000, because the old approach did not account for JSON overheard and was not always accurate in its truncation. I'm setting it much higher, so we should end up with more indexed content.

(I still want to leave a little wiggle-room here to account for any extra overhead in the API calls, or edge cases with byte boundaries.)